CMakeLists.txt

```diff
@@ -90,12 +90,14 @@ endif()
 # User-defined options.
 
 option (NN_STATIC_LIB "Build static library instead of shared library." OFF)
-option (NN_ENABLE_DOC "Enable building documentation." ON)
 option (NN_ENABLE_GETADDRINFO_A "Enable/disable use of getaddrinfo_a in place of getaddrinfo." ON)
 option (NN_TESTS "Build and run nanomsg tests" ON)
 option (NN_TOOLS "Build nanomsg tools" ON)
 option (NN_ENABLE_NANOCAT "Enable building nanocat utility." ${NN_TOOLS})
 
+set (NN_ENABLE_DOC "DEFAULT" CACHE STRING "Enable building documentation.")
+set_property (CACHE NN_ENABLE_DOC PROPERTY STRINGS "ON" "OFF" "DEFAULT")
```

Comment:
`yes` *might* work, but the nominal values are `OFF` or `ON` for options: https://cmake.org/cmake/help/v2.8.7/cmake.html#command:option

So, it *might* be limiting to anyone who has used non-standard boolean values, but the key here is that we're not boolean anymore -- it's an enumerated set of values.

The failure mode for such case mutating forward is fail-safe by throwing a fatal error with a hint.